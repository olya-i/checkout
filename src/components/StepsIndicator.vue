<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    totalSteps: Number,
    currentStep: Number,
  },
});
</script>

<template>
  <div class="steps-wrapper">
    <div
      v-for="step in totalSteps"
      :key="step"
      class="steps-wrapper__item"
      :class="{
        'steps-wrapper__item--filled': step - 1 <= Number(currentStep),
      }"
    ></div>
  </div>
</template>

<style lang="scss" scoped>
.steps-wrapper {
  display: flex;
  justify-content: space-between;
  margin-bottom: 44px;

  &__item {
    width: 30%;
    height: 12px;
    gap: 24px;
    border-radius: var(--border-radius--sm);
    background-color: var(--blue--light);
    background: linear-gradient(
        to left,
        var(--blue--light) 50%,
        var(--blue) 50%
      )
      right;
    background-size: 202%;
    transition: 0.3s ease-in-out;

    &--filled {
      background-color: var(--blue);
      background-position: left;
    }
  }
}
</style>
