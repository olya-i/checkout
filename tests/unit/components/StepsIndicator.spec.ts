import StepsIndicator from "../../../src/components/StepsIndicator.vue";
import { mount } from "@vue/test-utils";

describe("Mounted StepsIndicator", () => {
  it("renders correctly", () => {
    const wrapper = mount(StepsIndicator, {
      props: {
        totalSteps: 3,
        currentStep: 2,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
