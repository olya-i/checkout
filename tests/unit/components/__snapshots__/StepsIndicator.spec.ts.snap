// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted StepsIndicator renders correctly 1`] = `
<div
  class="steps-wrapper"
>
  
  <div
    class="steps-wrapper__item steps-wrapper__item--filled"
  />
  <div
    class="steps-wrapper__item steps-wrapper__item--filled"
  />
  <div
    class="steps-wrapper__item steps-wrapper__item--filled"
  />
  
</div>
`;
